src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INVALID_INDEX_MSG = "The index is invalid";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private class Node<T> {
+        private T val;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T val, Node<T> prev) {
```

Comment:
The node has a standard structure, sending to the next and previous element. I don't understand why you need to add such a constructor